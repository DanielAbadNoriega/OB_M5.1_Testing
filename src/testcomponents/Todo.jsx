import React, { useState } from "react";

const Todo = ({ todo }) => {
  const { text, id, completed } = todo;

  const [isCompleted, setIsCompleted] = useState(completed);

  const setCheck = () => {
    setIsCompleted(!isCompleted);
  };

  return (
    <div
      data-testid={`todo-test-${id}`}
      style={{
        display: "flex",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      <label>
        <input
          data-testid="checkbox"
          type="checkbox"
          checked={isCompleted}
          onChange={() => setCheck()}
        />
      </label>
      <p>{text}</p>
    </div>
  );
};

export default Todo;
