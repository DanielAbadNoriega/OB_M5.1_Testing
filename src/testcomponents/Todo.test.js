import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Todo from "./Todo";

describe("Todo component is working correctly.", () => {
  const todo = { id: 5, text: "hacer la compra", completed: true };
  render(<Todo todo={todo} />);
  const todoelement = screen.getByTestId(`todo-test-${todo.id}`);

  it("should render Todo component", () => {
    expect(todoelement).toBeInTheDocument();
  });
  it("should haver right text", () => {
    expect(todoelement).toHaveTextContent(todo.text);
  });
});

describe("Todo component should chech/uncheck completed/not completed todos", () => {
  const checkbox = screen.getByTestId("checkbox");

  it("should be checked", () => {
    expect(checkbox).toBeChecked();
  });

  it("matches snapshot", () => {
    const snapTodo = { id: 5, text: "hacer la compra 2", completed: true };
    const tree = renderer.create(<Todo todo={snapTodo} />);
    expect(tree).toMatchSnapshot();
  });
});
