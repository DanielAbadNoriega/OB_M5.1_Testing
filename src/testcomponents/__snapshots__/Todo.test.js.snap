// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Todo component should chech/uncheck completed/not completed todos matches snapshot 1`] = `
<div
  data-testid="todo-test-5"
  style={
    Object {
      "alignItems": "center",
      "display": "flex",
      "justifyContent": "center",
    }
  }
>
  <label>
    <input
      checked={true}
      data-testid="checkbox"
      onChange={[Function]}
      type="checkbox"
    />
  </label>
  <p>
    hacer la compra 2
  </p>
</div>
`;
